File: DSA/Aanidhay-Aggarwal/Day-10/Day10_q1.java
import java.util.Queue;
import java.util.LinkedList;

public class Main {
    public static Queue<Integer> reverseQueue(Queue<Integer> queue) {
        if (queue.isEmpty()) {
            return queue;
        }
        int front = queue.poll();
        queue = reverseQueue(queue);
        queue.add(front);
        return queue;
    }

    public static void main(String[] args) {
        Queue<Integer> queue = new LinkedList<>();
        queue.add(1);
        queue.add(2);
        queue.add(3);

        queue = reverseQueue(queue);

        System.out.println("Reversed queue: " + queue);
    }
}

File: DSA/Vyom-Singhal/Day10/Day10-q1.cpp
#include<iostream>
#include<queue>
using namespace std;

void reverseQueue(queue<int>& q) {
    if(q.empty()) return;
    int front = q.front();
    q.pop();
    reverseQueue(q);
    q.push(front);
}

void printQueue(queue<int> q) {
    while(!q.empty()) {
        cout << q.front() << " ";
        q.pop();
    }
}

int main() {
    queue<int> q;
    q.push(1);
    q.push(2);
    q.push(3);
    printQueue(q);
    cout << endl;
    reverseQueue(q);
    printQueue(q);
}
